dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/category/DefaultCategoryManager.java

```diff
@@ -75,45 +75,62 @@ public void addAndPruneOptionCombos( CategoryCombo categoryCombo )
 
         boolean modified = false;
 
-        for ( CategoryOptionCombo optionCombo : generatedOptionCombos )
-        {
-            if ( !persistedOptionCombos.contains( optionCombo ) )
-            {
-                categoryCombo.getOptionCombos().add( optionCombo );
-                categoryService.addCategoryOptionCombo( optionCombo );
-
-                log.info( "Added missing category option combo: " + optionCombo + " for category combo: " + categoryCombo.getName() );
-                modified = true;
-            }
-        }
 
         Iterator<CategoryOptionCombo> iterator = persistedOptionCombos.iterator();
 
         while ( iterator.hasNext() )
         {
-            CategoryOptionCombo optionCombo = iterator.next();
+            CategoryOptionCombo persistedOptionCombo = iterator.next();
+
+            boolean isDelete = true;
+
+            for ( CategoryOptionCombo c : generatedOptionCombos )
+            {
+                if ( c.equals( persistedOptionCombo ) || c.getUid() == persistedOptionCombo.getUid() )
```

Comment:
Use .equals instead of == for string comparison

Reply:
Fixed